api/tools/src/main/java/org/pmiops/workbench/tools/LoadDataDictionary.java

```diff
@@ -41,41 +43,48 @@ public CommandLineRunner run(
 
       for (Resource resource : getDataDictionaryExportFiles()) {
         InputStream is = resource.getInputStream();
+        ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
+        SimpleDateFormat df = new SimpleDateFormat("dd-MM-yyyy hh:mm:ss");
+        mapper.setDateFormat(df);
 
-        BufferedReader reader = new BufferedReader(new InputStreamReader(is));
-        Timestamp definedTime = new Timestamp(Long.parseLong(reader.readLine()) * 1000);
+        mapper.registerModule(new KotlinModule());
+        DataDictionary dd = mapper.readValue(is, DataDictionary.class);
 
-        String line;
-        while ((line = reader.readLine()) != null) {
-          String[] fields = line.split(":");
+        Timestamp newEntryDefinedTime = dd.getMeta_data()[0].getCreated_time();
 
-          String relevantOmopTable = fields[0];
-          String fieldName = fields[1];
+        CdrVersion cdrVersion = cdrVersionDao.findByName(dd.getMeta_data()[0].getCdr_version());
 
+        if (cdrVersion == null) {
+          // Skip over Data Dictionaries for CDR Versions not in the current environment
+          continue;
+        }
+
+        for (AvailableField field : dd.getTransformations()[0].getAvailable_fields()) {
           DataDictionaryEntry entry =
               dataDictionaryEntryDao.findByRelevantOmopTableAndFieldNameAndCdrVersion(
-                  relevantOmopTable, fieldName, defaultCdrVersion);
+                  field.getRelevant_omop_table(), field.getField_name(), cdrVersion);
```

Comment:
nit: can we not set up the find methods in the dao class to return an `Optional<DataDictionaryEntry>`? Seems like that could consolidate both null checks.